controller/src/classes/Controller.ts

```diff
@@ -0,0 +1,165 @@
+import { BaseSubject, Permission, ResourcePermissions } from "../types";
+import { EnforceKeyMatchResolver, IController, IPermissionManager, IStore, SubjectKey, SubjectType } from "../types/modules";
+import { Mutex } from "./utils/Mutex";
+
+export class Controller<T extends Record<keyof T, BaseSubject<keyof T & string>>> extends Mutex implements IController<T> {
+    private store: IStore
+    private subjectResolvers: EnforceKeyMatchResolver<T>
+    private permissionManager: IPermissionManager<T>
+
+    constructor(store: IStore, subjectResolvers: EnforceKeyMatchResolver<T>, pm: IPermissionManager<T>) {
+        super();
+        this.store = store;
+        this.permissionManager = pm;
```

Comment:
I would expect, for each 'permission type' to have a permission function --> you could have a webId permission function (eg from inrupt), and a qualifier permission function (e.g. from KNoWS)